aks-node-controller/proto/aksnodeconfig/v1/config.proto

```diff
@@ -157,4 +157,10 @@ message Configuration {
 
   // Represents localdns profile for agentpool.
   LocalDnsProfile local_dns_profile = 41;
+
+  // Specifies whether to disable ssh public key authentication on the VM node
+  // This is labeled as optional (explicit presence) so that we know whether it's set or not.
+  // If it's not set, the default value will be nil, but will be set to false on the VHD.
+  // That is, the default behavior is to enable ssh public key authentication.
+  optional bool disable_pubkey_auth = 42;
```

Comment:
IIUC, if you just want the default value to be `false`, you don't need to set it as `optional`. 
`explicit presence` is only needed if you want to know if the variable is in `unset` state and you have additional logic to handle it. In this case, it seems you don't need it. (Please correct me if I am wrong)

Without `optional`, if user sets it to `true`, you will get a `true`. If user sets it to `false` or unsets it, you will get `false`.

Reply:
yes. If that's needed or will be needed. No harm to keep it as `optional`. Just wanted to make it clear.